app/models/test_track/session.rb

```diff
@@ -114,8 +114,16 @@ def notify_unsynced_assignments!
       ##
       # This block creates an unbounded number of threads up to 1 per request.
       # This can potentially cause issues under high load, in which case we should move to a thread pool/work queue.
+      original_store = RequestStore.store
       Thread.new do
```

Comment:
You can do `Thread.new(RequestStore.store) do |original_store|` to make a local copy only in the child thread and avoid accessing any shared memory.